src/transformers/models/fast/modeling_fast.py

```diff
@@ -0,0 +1,932 @@
+            "ops_order": self.ops_order,
+        }
+
+    @staticmethod
+    def build_from_config(config):
+        raise NotImplementedError
+
+    def get_flops(self, x):
+        raise NotImplementedError
+
+    @staticmethod
+    def is_zero_layer():
+        return False
+
+
+class ConvLayer(nn.Module):
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        kernel_size=3,
+        stride=1,
+        dilation=1,
+        groups=1,
+        bias=False,
+        has_shuffle=False,
+        use_bn=True,
+        act_func="relu",
+        dropout_rate=0,
+        use_act=True,
+    ):
+        super().__init__()
+
+        self.kernel_size = kernel_size
+        self.stride = stride
+        self.dilation = dilation
+        self.groups = groups
+        self.bias = bias
+        self.has_shuffle = has_shuffle
+        self.act_func = act_func
+
+        padding = get_same_padding(self.kernel_size)
+        if isinstance(padding, int):
+            padding *= self.dilation
+        else:
+            padding[0] *= self.dilation
+            padding[1] *= self.dilation
+
+        self.conv = nn.Conv2d(
+            in_channels,
+            out_channels,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            dilation=dilation,
+            groups=groups,
+            bias=bias,
+        )
+        self.bn = nn.Identity()
+        if use_bn:
+            self.bn = nn.BatchNorm2d(out_channels)
+
+        self.act = nn.Identity()
+        if use_act:
+            act = build_activation(self.act_func, True)
+            if act is not None:
+                self.act = act
+
+    def forward(self, x):
+        if self.training:
+            if hasattr(self, "fused_conv"):
+                delattr(self, "fused_conv")
+            x = self.conv(x)
+            x = self.bn(x)
+            return self.act(x)
+        else:
+            if not hasattr(self, "fused_conv"):
+                setattr(self, "fused_conv", self.fuse_conv_bn(self.conv, self.bn))
+            x = self.fused_conv(x)
+            if self.act is not None:
+                x = self.act(x)
+            return x
+
+    def fuse_conv_bn(self, conv, bn):
+        """During inference, the functionary of batch norm layers is turned off but
+        only the mean and var alone channels are used, which exposes the chance to fuse it with the preceding conv
+        layers to save computations and simplify network structures."""
+        if isinstance(bn, nn.Identity):
+            return conv
+        conv_w = conv.weight
+        conv_b = conv.bias if conv.bias is not None else torch.zeros_like(bn.running_mean)
+
+        factor = bn.weight / torch.sqrt(bn.running_var + bn.eps)
+        conv.weight = nn.Parameter(conv_w * factor.reshape([conv.out_channels, 1, 1, 1]))
+        conv.bias = nn.Parameter((conv_b - bn.running_mean) * factor + bn.bias)
+        return conv
+
+
+class RepConvLayer(nn.Module):
```

Comment:
```suggestion
class FastRepConvLayer(nn.Module):
```
What does rep mean? Could have a better name